static/compiler-picker.ts

```diff
@@ -0,0 +1,209 @@
+        this.onCompilerChange = onCompilerChange;
+        this.eventHub.on('compilerFavoriteChange', this.onCompilerFavoriteChange, this);
+        this.tomSelect = null;
+        if (compilerIsVisible) {
+            this.compilerIsVisible = compilerIsVisible;
+        } else {
+            this.compilerIsVisible = () => true;
+        }
+
+        this.initialize(langId, compilerId);
+    }
+
+    close() {
+        // Quick note while I'm here: This function is never called. It probably should be. The conformance view
+        // bypasses this function and does compilerEntry.picker.tomSelect.close(); manually. This function is the
+        // only time this.tomSelect can be null, might be nice if we can get rid of that.
+        this.eventHub.unsubscribe();
+        if (this.tomSelect)
+            this.tomSelect.destroy();
+        this.tomSelect = null;
+    }
+
+    initialize(langId: string, compilerId: string) {
+        this.lastLangId = langId;
+        this.lastCompilerId = compilerId;
+
+        this.tomSelect = new TomSelect(this.domNode as any /*TODO*/, {
+            sortField: this.compilerService.getSelectizerOrder(),
+            valueField: 'id',
+            labelField: 'name',
+            searchField: ['name'],
+            placeholder: '🔍 Select a compiler...',
+            optgroupField: '$groups',
+            optgroups: this.getGroups(langId),
+            lockOptgroupOrder: true,
+            options: this.getOptions(langId, compilerId),
+            items: compilerId ? [compilerId] : [],
+            dropdownParent: 'body',
+            closeAfterSelect: true,
+            plugins: ['dropdown_input'],
+            maxOptions: 1000,
+            onChange: (val) => {
+                if (val) {
+                    ga.proxy('send', {
+                        hitType: 'event',
+                        eventCategory: 'SelectCompiler',
+                        eventAction: val,
+                    });
+                    this.onCompilerChange(val);
+                    this.lastCompilerId = val as any; // TODO
+                }
+            },
+            duplicates: true,
+            render: {
+                option: (data, escape) => {
+                    const isFavoriteGroup = data.$groups.indexOf(CompilerPicker.favoriteGroupName) !== -1;
+                    const extraClasses = isFavoriteGroup ? 'fas fa-star fav' : 'far fa-star';
+                    return '<div class="d-flex"><div>' + escape(data.name) + '</div>' +
+                        '<div title="Click to mark or unmark as a favorite" class="ml-auto toggle-fav">' +
+                        '<i class="' + extraClasses + '"></i>' +
+                        '</div>' +
+                        '</div>';
+                },
+            },
+        });
+
+        $(this.tomSelect.dropdown_content).on('click', '.toggle-fav', evt => {
+            evt.preventDefault();
+            evt.stopPropagation();
+
+            if(this.tomSelect) {
```

Comment:
Linter is going to complain about this one missing spaces

Reply:
haha good catch 😅